src/core/manager-json.c

```diff
@@ -0,0 +1,181 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <sys/prctl.h>
+
+#include "build.h"
+#include "confidential-virt.h"
+#include "json-util.h"
+#include "manager-json.h"
+#include "manager.h"
+#include "rlimit-util.h"
+#include "syslog-util.h"
+#include "taint.h"
+#include "version.h"
+#include "virt.h"
+#include "watchdog.h"
+
+static int rlimit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        struct rlimit *rl = userdata, buf = {};
```

Comment:
hmm, i'd do this the other way round, i.e. `const struct rlimit *rl = userdata; struct rlimit *buf = {};`. And then when `rl` is not null, set `rl = &buf;` and initialize `buf`. And then use `rl` all the way to the end.